_posts/2019-09-12-API-first-development-with-OpenAPI-or-Swagger.md

```diff
@@ -0,0 +1,549 @@
+---
+layout: post
+authors: [nick_van_hoof]
+title: 'API first development with OpenAPI/Swagger'
+image: /img/2019-09-12-API-first-development-with-OpenAPI-or-Swagger/featured-image.png
+tags: [API, cloud]
+category: Cloud
+comments: true
+---
+
+# Table of content
+
+1.[API first development - Why, how and what](#api-first-development---why-how-and-what)  
+2.[API design: an example](#api-design-an-example)  
+3.[API first development with OpenAPI/Swagger](#api-first-development-with-openapiswagger)  
+4.[Hosting your visualisations](#hosting-your-visualisations)  
+5.[Integrating with Postman](#integrating-with-postman)  
+6.[OpenAPI Generator: generate API compliant code](#openapi-generator-generate-api-compliant-code)  
+7.[Integrating with your build process: maven or gradle plugin](#integrating-swaggeropenapi-with-your-build-process-maven-or-gradle-plugin)  
+8.[Serverless on AWS: OpenAPI, API Gateway Lambda and SAM](#serverless-on-aws-openapi-api-gateway-lambda-and-sam)  
+9.[Springfox](#code-first-with-springfox)   
+10.[Conclusion](#conclusion)
+
+
+
+# Intro
+* I'll start of with a lecture about API first development and it's advantages.([Jump to section](#api-first-development---why-how-and-what))
+* We will discover how to visualize our API specs. ([Jump to section](#api-first-development-with-openapiswagger))  
+* Generate a postman collection from your OpenAPI definition. ([Jump to section](#integrating-with-postman))   
+* Next I'll show you how to generate code that is completely compliant with your API specs. ([Jump to section](#openapi-generator-generate-api-compliant-code))
+* We then dive into how to integrate this in your build process by using maven or gradle plugins. ([Jump to section](#integrating-with-your-build-process-maven-or-gradle-plugin))
+* To finish it up I'll demonstrate how to use it in a cloud native serverless product with `AWS SAM` and `AWS Lambda`. ([Jump to section](#serverless-on-aws-openapi-api-gateway-lambda-and-sam))
+* Wrapping it up with a conclusion.
+
+# API first development - Why, how and what
+> Great communication is key to great software engineering.
```

Comment:
A little intro about the concept of API economy would be nice. 
For what stands the abbreviation.